src/test/java/com/mongodb/jdbc/MongoSQLResultSetTest.java

```diff
@@ -285,6 +320,186 @@ void testGetStringValues() throws Exception {
+                mongoSQLResultSetAllTypes.getObject(ALL_OBJECT_ID_COL_LABEL).toString());
+        assertEquals("true", mongoSQLResultSetAllTypes.getObject(ALL_BOOL_COL_LABEL).toString());
+        assertEquals(
+                "2020-12-25 12:13:14.0",
+                mongoSQLResultSetAllTypes.getObject(ALL_DATE_COL_LABEL).toString());
+        assertEquals(
+                "{\"$regularExpression\": {\"pattern\": \"abc\", \"options\": \"i\"}}",
+                mongoSQLResultSetAllTypes.getObject(ALL_REGEX_COL_LABEL).toString());
+        assertEquals(
+                "{\"$code\": \"javascript\"}",
+                mongoSQLResultSetAllTypes.getObject(ALL_JAVASCRIPT_COL_LABEL).toString());
+        assertEquals(
+                "{\"$symbol\": \"sym\"}",
+                mongoSQLResultSetAllTypes.getObject(ALL_SYMBOL_COL_LABEL).toString());
+        assertEquals(
+                "{\"$code\": \"code\", \"$scope\": {\"x\": 1}}",
+                mongoSQLResultSetAllTypes
+                        .getObject(ALL_JAVASCRIPT_WITH_SCOPE_COL_LABEL)
+                        .toString());
+        assertEquals("3", mongoSQLResultSetAllTypes.getObject(ALL_INT_COL_LABEL).toString());
+        assertEquals(
+                "{\"$timestamp\": {\"t\": 1412180887, \"i\": 1}}",
+                mongoSQLResultSetAllTypes.getObject(ALL_TIMESTAMP_COL_LABEL).toString());
+        assertEquals("5", mongoSQLResultSetAllTypes.getObject(ALL_LONG_COL_LABEL).toString());
+        assertEquals("21.2", mongoSQLResultSetAllTypes.getObject(ALL_DECIMAL_COL_LABEL).toString());
+        assertEquals(
+                "{\"$minKey\": 1}",
+                mongoSQLResultSetAllTypes.getObject(ALL_MIN_KEY_COL_LABEL).toString());
+        assertEquals(
+                "{\"$maxKey\": 1}",
+                mongoSQLResultSetAllTypes.getObject(ALL_MAX_KEY_COL_LABEL).toString());
+
+        // Note that the Java driver still outputs the legacy representation for DBPointer, as
+        // opposed to the new standard representation: { $dbPointer: { $ref: <namespace>, $id: <oid> } }.
+        // This is sufficient for our purposes, though.
+        assertEquals(
+                "{\"$ref\": \"db2\", \"$id\": {\"$oid\": \""
+                        + ALL_OBJECT_ID_VAL.toString()
+                        + "\"}}",
+                mongoSQLResultSetAllTypes.getObject(ALL_DB_POINTER_COL_LABEL).toString());
+
+        // Note that getObject() returns null for NULL and UNDEFINED BSON values, so we check
+        // manually that their stringification returns what is expected.
+        assertNull(mongoSQLResultSetAllTypes.getObject(ALL_UNDEFINED_COL_LABEL));
+        assertNull(new MongoSQLValue(new BsonUndefined()).toString());
+        assertNull(mongoSQLResultSetAllTypes.getObject(ALL_NULL_COL_LABEL));
+        assertNull(new MongoSQLValue(new BsonNull()).toString());
+
+        // Test getObject().toString() is valid relaxed extended JSON
+        // TODO
```

Comment:
I don't think that we are testing that the value are valid extended JSON value which are strictly matching the extended JSON spec. We are only testing that both getString and getObject.toString() are the same. If we were testing that the values are valid extended JSON, then you would see a failure for the Timestamp.
Try this out :
long unixTime = System.currentTimeMillis();
BsonDateTime timestamp = new BsonDateTime(unixTime);
Date date = new Date(timestamp.getValue());
extendedJsonVal = printExtendedJsonValue(timestamp, JsonMode.RELAXED);
System.out.println(dateFormat.format(date));
System.out.println(extendedJsonVal);

You'll notice that the 2 string representations are different.

Also, the date from timestamp.getValue() is also not in UTC even though the timezone implies it is, while the extended Json value is in UTC. I'll file a separate ticket for this.

Reply:
Yea sorry, noted on another thread: I didn't know we wanted to change getString() for BSON types that correspond to non-Types.OTHER JDBC types. However, even after we update to make sure getString()/getObject().toString() return extended json representations for all types, I'm not sure what additional coverage this additional test will provide.

We'll have the following assertions for every bson type:
```
assertEquals("{$oid: ...}", rs.getString(OID_COL));
assertEquals("1.2", rs.getString(DOUBLE_COL));
assertEquals("{$date: ...}, rs.getString(DATE_COL));
... // etc

assertEquals("$oid: ...}", rs.getObject(OID_COL).toString());
assertEquals("1.2", rs.getObject(DOUBLE_COL).toString());
assertEquals("{$date: ...}, rs.getObject(DATE_COL).toString());
// etc
```
Two assertions for each bson type where the expected string is the extended json representation. Since the expected string is already the extended json string representation, isn't that sufficient to show that the output of getString()/getObject().toString() is _valid_ extended json? It seems the additional test to ensure the output is valid would really just be testing that our expected strings in the other tests are valid.

Reply:
It would if we could be 100% sure that the string we put there is the correct extended json representation and using an extended Json compatible encoder works to reconstruct the BsonValue. As you can see, it is not the case for date_time and Decimal128 right now. Let's see after we discuss if we are indeed using the extended representation for every data type or if the others in the team also thought this was just for data types mapping to Other. After that, we can see how to make maybe change this test slightly to cover both it if needed.